lemmy_structs/src/comment.rs

```diff
@@ -76,3 +79,41 @@ pub struct GetComments {
 pub struct GetCommentsResponse {
   pub comments: Vec<CommentView>,
 }
+
+#[derive(Serialize, Deserialize)]
+pub struct CreateCommentReport {
+  pub comment_id: i32,
+  pub reason: String,
+  pub auth: String,
+}
+
+#[derive(Serialize, Deserialize, Clone)]
+pub struct CreateCommentReportResponse {
+  pub success: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ResolveCommentReport {
+  pub report_id: i32,
+  pub resolved: bool,
+  pub auth: String,
+}
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct ResolveCommentReportResponse {
+  pub report_id: i32,
+  pub resolved: bool,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct ListCommentReports {
+  pub page: Option<i64>,
+  pub limit: Option<i64>,
+  pub community: Option<i32>,
```

Comment:
We'll have to remember to put a comment here in the docs that if a single community isn't given, then it lists all your modded communities reports.

Reply:
Would be nice to have it commented in the code as well.

Reply:
bump